src/devices/Ak8963/Ak8963.cs

```diff
@@ -0,0 +1,258 @@
+        /// Check if the version is the correct one (0x48)
+        /// </summary>
+        /// <returns>Returns true if the version match</returns>
+        public bool CheckVersion()
+        {
+            return ReadByte(Register.WIA) == 0x48;
+        }
+
+        /// <summary>
+        /// Read the magnetomer and can wait for new data to be present
+        ///         +X
+        ///  \  |  /
+        ///   \ | /
+        ///    \|/
+        ///    /|\
+        ///   / | \
+        ///  /  |  \
+        ///    +Z   +Y
+        /// </summary>
+        /// <param name="waitForData">true to wait for new data</param>
+        /// <returns>The data from the magnetometer</returns>
+        public Vector3 ReadMagnetometer(bool waitForData = false)
+        {
+            Span<byte> rawData = stackalloc byte[6];
+            // Wait for a data to be present
+            if (waitForData)
+            {
+                while (!DataReady)
+                    ;
+            }
+            ReadByteArray(Register.HXL, rawData);
+            // In continous mode, make sure to read the ST2 data to clear up
+            if ((_measurementMode == MeasurementMode.ContinousMeasurement100Hz) ||
+                (_measurementMode == MeasurementMode.ContinousMeasurement8Hz))
+                ReadByte(Register.ST2);
+
+            Vector3 magneto = new Vector3();
+            magneto.X = BinaryPrimitives.ReadInt16LittleEndian(rawData);
+            magneto.Y = BinaryPrimitives.ReadInt16LittleEndian(rawData.Slice(2));
+            magneto.Z = BinaryPrimitives.ReadInt16LittleEndian(rawData.Slice(4));
+            if (OutputBitMode == OutputBitMode.Output16bit)
+            {
+                // From the documentation range is from 32760 which does represent 4912 µT
+                magneto *= 4912.0f / 32760.0f;
+            }
+            else
+            {
+                magneto *= 4912.0f / 8192.0f;
+            }
+            return magneto;
+
+        }
+
+        /// <summary>
+        /// Get the Magetometer data and wait for them to be ready
+        ///         +X
+        ///  \  |  /
+        ///   \ | /
+        ///    \|/
+        ///    /|\
+        ///   / | \
+        ///  /  |  \
+        ///    +Z   +Y
+        /// </summary>
+        public Vector3 Magnetometer => ReadMagnetometer(true);
+
+        /// <summary>
+        /// Get or set the device self test mode. 
+        /// If set to true, this creates a magnetic field
+        /// Once you read it, you will have the results of the self test
+        /// 14-bit output(BIT=“0”)  
+        ///          | HX[15:0]        | HY[15:0]        | HZ[15:0] 
+        /// Criteria | -50 =< HX =< 50 | -50 =< HY =< 50 | -800 =< HZ =< -200 
+        /// 16-bit output(BIT=“1”)  
+        ///          | HX[15:0]          | HY[15:0]          | HZ[15:0] 
+        /// Criteria | -200 =< HX =< 200 | -200 =< HY =< 200 | -3200 =< HZ =< -800 
+        /// </summary>
+        public bool SelfTest
```

Comment:
I think this should be wrapped as a function:

void SelfTest()
{
  EnableSelfTest();
  // do checks, throw if something is wrong (or return true/false)
  DisableSelfTest();
}

Reply:
The point with the self test is that you can do other things. It creates a magnetic field. Which you can use to calibrate other elements. And read as well the data. That's the reason why I implemented it like this. I can add as well a function with returns the magnetometer data. Let me know what you think.

Reply:
Perhaps rename this to `MageneticFieldGeneratorEnabled`